plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesSplitSource.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.functions.tablechanges;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Closer;
+import io.trino.plugin.iceberg.IcebergFileFormat;
+import io.trino.plugin.iceberg.PartitionData;
+import io.trino.spi.SplitWeight;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorSplitSource;
+import io.trino.spi.type.DateTimeEncoding;
+import org.apache.iceberg.AddedRowsScanTask;
+import org.apache.iceberg.ChangelogScanTask;
+import org.apache.iceberg.DeletedDataFileScanTask;
+import org.apache.iceberg.IncrementalChangelogScan;
+import org.apache.iceberg.PartitionSpecParser;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.CloseableIterator;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public class TableChangesSplitSource
+        implements ConnectorSplitSource
+{
+    private final Table icebergTable;
+    private final IncrementalChangelogScan tableScan;
+    private final Closer closer = Closer.create();
+
+    private CloseableIterable<ChangelogScanTask> changelogScanIterable;
+    private CloseableIterator<ChangelogScanTask> changelogScanIterator;
+
+    public TableChangesSplitSource(
+            Table icebergTable,
+            IncrementalChangelogScan tableScan)
+    {
+        this.icebergTable = requireNonNull(icebergTable, "table is null");
+        this.tableScan = requireNonNull(tableScan, "tableScan is null");
+    }
+
+    @Override
+    public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
+    {
+        if (changelogScanIterable == null) {
+            try {
+                this.changelogScanIterable = closer.register(tableScan.planFiles());
+                this.changelogScanIterator = closer.register(changelogScanIterable.iterator());
```

Comment:
in IcebergSplitSource each ScanTask is further split into an iterator of smaller splits according to target split size. Does it make sense to keep the same scan logic here?

Reply:
Absolutely, I just got thrown off that `ChangelogScanTask` doesn't implement `SplittableScanTask` so it isn't exactly like the main read path. Did you figure it out in your version?

Reply:
Ah, I see. Let me try to clean up the interface definitions in Iceberg core to clear that up but that'll work for now.